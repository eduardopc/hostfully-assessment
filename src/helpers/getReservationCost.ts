import { differenceInDays } from "date-fns";
import { GENERAL } from "languages";

import { PlaceSelectOption } from "types";

type GetReservationCostProps = {
  dates?: string[];
  pricePerDay: PlaceSelectOption["pricePerDay"];
};

export const getReservationCost = ({
  dates,
  pricePerDay,
}: GetReservationCostProps): string => {
  if (!dates || !pricePerDay) return GENERAL.remake_reservation;

  const days = differenceInDays(dates[1], dates[0]);

  return `$${days * pricePerDay}`;
};
