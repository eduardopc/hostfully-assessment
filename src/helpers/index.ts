export { isBookingDateRangeAvailable } from "./isBookingDateRangeAvailable";
export { showToastMessage } from "./toast";
