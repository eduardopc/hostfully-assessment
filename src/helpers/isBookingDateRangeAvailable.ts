import { isWithinInterval } from "date-fns";

import { BookingDataProps } from "contexts/types";

export const isBookingDateRangeAvailable = (
  from: Date,
  to: Date,
  bookings: BookingDataProps[]
): boolean => {
  return !bookings.some(
    (booking) =>
      isWithinInterval(from, {
        start: booking.date[0].from,
        end: booking.date[0].to,
      }) ||
      isWithinInterval(to, {
        start: booking.date[0].from,
        end: booking.date[0].to,
      })
  );
};
