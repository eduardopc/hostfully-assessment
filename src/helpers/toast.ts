import { toast, TypeOptions } from "react-toastify";

type ShowToastMessage = {
  message: string;
  type: TypeOptions;
  onOpen?: () => void;
};

export const showToastMessage = ({
  message,
  type,
  onOpen,
}: ShowToastMessage) => {
  toast[type](message, {
    position: "top-right",
    autoClose: 5000,
    hideProgressBar: true,
    closeOnClick: true,
    pauseOnHover: true,
    draggable: false,
    theme: "colored",
    onOpen: () => {
      onOpen?.();
    },
  });
};
