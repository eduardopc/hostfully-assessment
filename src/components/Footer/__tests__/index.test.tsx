import { screen } from "@testing-library/react";
import { renderWithTheme } from "utils/tests/helpers";

import { Footer } from "..";

describe("<Footer />", () => {
  it("should render 3 columns topics", () => {
    const { container } = renderWithTheme(<Footer />);

    expect(screen.getByRole("heading", { name: /contact us/i })).toBeDefined();
    expect(screen.getByRole("heading", { name: /follow us/i })).toBeDefined();
    expect(screen.getByRole("heading", { name: /location/i })).toBeDefined();

    expect(container.firstChild).toMatchSnapshot();
  });
});
