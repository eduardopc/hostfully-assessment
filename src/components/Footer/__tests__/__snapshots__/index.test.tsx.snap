// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Footer /> > should render 3 columns topics 1`] = `
<footer
  class="sc-gTRrcP hjDaIy"
>
  <div
    class="sc-ewnpUy fVmliq"
  >
    <div
      class="sc-fFGkvd cBhQVS"
    >
      <h2
        class="sc-bgqQPU iNiuGW"
        color="black"
      >
        Contact Us
      </h2>
      <a
        href="mailto:sample@mail.com"
      >
        sample@mail.com
      </a>
    </div>
    <div
      class="sc-fFGkvd cBhQVS"
    >
      <h2
        class="sc-bgqQPU iNiuGW"
        color="black"
      >
        Follow us
      </h2>
      <nav
        aria-labelledby="social media"
      >
        <a
          href="https://www.instagram.com/hostfully_/"
          rel="noopenner, noreferrer"
          target="_blank"
        >
          Instagram
        </a>
        <a
          href="https://www.linkedin.com/company/hostfully-inc-/"
          rel="noopenner, noreferrer"
          target="_blank"
        >
          LinkedIn
        </a>
      </nav>
    </div>
    <div
      aria-labelledby="footer-contact"
      class="sc-fFGkvd cBhQVS"
    >
      <h2
        class="sc-bgqQPU iNiuGW"
        color="black"
      >
        Location
      </h2>
      <span>
        Lorem ipsum dolor sit.
      </span>
    </div>
  </div>
  <div
    class="sc-dicjmM eOshWU"
  >
    Developed by 
    <strong>
      Hostfully
    </strong>
     - 2024 Â© All rights reserved
  </div>
</footer>
`;
